openpype/hosts/maya/plugins/publish/extract_look.py

```diff
@@ -429,9 +429,14 @@ def process_resources(self, instance, staging_dir):
                 # node doesn't have color space attribute
                 color_space = "Raw"
             else:
-                if files_metadata[source]["color_space"] == "Raw":
+                try:
+                    if files_metadata[source]["color_space"] == "Raw":
                     # set color space to raw if we linearized it
-                    color_space = "Raw"
+                        color_space = "Raw"
+                except KeyError:
+                    # set color space to Raw if its attribute is raw.
+                    if cmds.getAttr(color_space_attr) == "Raw":
+                        color_space = "Raw"
```

Comment:
These lines of code actually wouldn't be needed because we only enter the `else:` clause of the `try.. except.. else` earlier when the `try..` succeeds without a captured error. As such `color_space = "Raw"` already.

---

However, I think we're solving this the wrong way anyway and we're "patching" an issue that originates elsewhere.

I have the feeling as well that we should instead be solving the issue as to why `source` doesn't work with `aiImage` to begin with. This actually sounds more like the resource isn't being collected correctly to begin with and might thus in the future run into other issues that might expect the resource itself to be collected correctly.

The question thus becomes why does a `file` node with UDIM sequence get collected correctly as a resource and why does `aiImage` with UDIM does not.

Reply:
I actually try removing the condition and just set` else: color_space = "Raw"` before but I am not very certain of the condition being relevant to associate any data. But after you mention about the "wrong way" and issues which originates elsewhere, I'd prefer more concise solution for this issue instead of being cliche with both exception handling and conditon.
I agree that there might be running into other issues which is related to collecting resources if the resource isn't being collected correctly

Reply:
You might have misinterpreted but now the code is actually overriding `color_space` whenever it wasn't "Raw" to begin with - the only example I described that would be equivalent is the one of:

```python
                    # set color space to Raw if its attribute is raw.
                    if cmds.getAttr(color_space_attr) == "Raw":
                        color_space = "Raw"
```
If the attribute was set as "Raw" then `color_space = "Raw"` was already the case.
The current PR state after https://github.com/pypeclub/OpenPype/pull/3628/commits/38c35a87dea322e8fb81179cb40abd0549a905b7 will break other things.

Anyway, it doesn't matter much since what we're looking to solve anyway is why  `files_metadata[source]["color_space"]` didn't have the `source` entry to begin with. 

We'll need to look into where the `files_metadata[source]` entry is generated in the first place. And we'll need to make sure that the source is actually correctly added there already so that the original logic below would actually work as expected:
```python
                if files_metadata[source]["color_space"] == "Raw":
                    # set color space to raw if we linearized it
                    color_space = "Raw"
```

Reply:
Hi Roy. I took a look at how the `files_metadata `is generated as I see it is implemented as a dictionary set with a normalized path with `colors_space `inside it. It is generated through two loops(one is `resource `iterating through `resources`, the other is `f `iterating through `resources[files]`)
`        files_metadata = {}
        for resource in resources:
            # Preserve color space values (force value after filepath change)
            # This will also trigger in the same order at end of context to
            # ensure after context it's still the original value.
            color_space = resource.get("color_space")

            for f in resource["files"]:
                files_metadata[os.path.normpath(f)] = {
                    "color_space": color_space}`

And then I saw this one with a `filepath `iterating through `files_metadata`:

`            if do_maketx and files_metadata[filepath]["color_space"].lower() == "srgb":  # noqa: E501
                linearize = True
                # set its file node to 'raw' as tx will be linearized
                files_metadata[filepath]["color_space"] = "Raw"
`

I also take a look at where ` [source] ` is generated
`   for resource in resources:
            source = os.path.normpath(resource["source"])
            if source not in destinations:
                # Cache destination as source resource might be included
                # multiple times
                destinations[source] = self.resource_destination(
                    instance, source, do_maketx
                )
`
I guess we need to figure out where resource["source"] from. I guess it's probably from something related to the tx options.